### DISCUSSION TITLE
Default device reporting interval

### DISCUSSION THREAD

**Discussion by Morphy99** [Koenkk/zigbee2mqtt#5708] [General]
I've had a look around and couldn't find anything specific that states what the default reporting for a device is? I'm guessing for battery powered devices this is quite important in terms of power saving. Looking at the frontend reporting tab, all devices seem to have Min rep interval = 60, Max rep interval = 3600 where I think they are actually defaulted to something different. 

Is there a way to read this value or is it just a case of timing the frequency of the messages manually? Maybe we could add this info to the device page on the wiki?

**Comment by Koenkk** [comment]
A device does not have a default reporting interval, all reporting intervals are configured from z2m, you can find them here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/lib/reporting.js

**Reply by Morphy99** [reply]
Wow OK that's surprising I assumed the manufacturer would set the interval, especially for battery powered devices as different models would have different battery consumptions. 
So the report interval is set by Z2M depending on what attributes are available on the device? And only those attributes listed in reporting.js are configured by default?

Are there any docs or threads I can look at to learn more about this? Couldn't see anything on the wiki.
Looking at reporting.js for example, this attribute reporting is every 10 seconds? Also what does the figure after localtemp represent?

```},
    thermostatTemperature: async (endpoint, overrides) => {
        const p = payload('localTemp', 0, repInterval.HOUR, 10, overrides);
        await endpoint.configureReporting('hvacThermostat', p);```

**Reply by Koenkk** [reply]
What "figure" do you mean?

The default reportings being set-up can be found in the `configure` method for each device (e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/c8ca92c7f04e6c27cecd0ce1514f09f04dd9abd1/devices.js#L401)

**Reply by Morphy99** [reply]
Sorry. What I meant by figure is probably better worded as value. The "0" here:
const p = payload('localTemp', **0**, repInterval.HOUR, 10, overrides);

I can't see any values for reportings in devices.js?

I was guessing the repInterval.HOUR, **10** was a reporting every ten seconds?

Bare with me, I'm just trying to understand the code without any coding knowledge :smile:

**Reply by Koenkk** [reply]
`payload('localTemp', 0, repInterval.HOUR, 10, overrides);` means, min report interval = 0 seconds, max report interval = 1 hour, minimal reportable change = 10 units (local temp 1 unit = 0.1 c). For more info about the configure reporting you can refer to the ZCL: https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/Zigbee%20Cluster%20Library%20Specification%20v7.pdf

**Reply by aharrys** [reply]
Hy @Koenkk could you  re - send link https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/Zigbee%20Cluster%20Library%20Specification%20v7.pdf. 

the file is 404  can't be find
thanks

**Reply by spry-salt** [reply]
Sorry to resurrect an old thread but I have the same problem now with current versions of Home Assistant, z2m and SmartThings Multipurpose Sensor (2018) devices, model IM6001-MPP01. They're reporting every minute now and are burning through their batteries. 

From the z2m debug log it looks like they're reporting every minute due to a change in their LQI value - all other values look like they're the same.

(EDIT: I've also tried changing the reporting intervals for the accelerometer settings but z2m flatly refuses to do that even though I have the device sitting next to the coordinator and it's reporting every minute without problems:
`z2m: Request 'zigbee2mqtt/bridge/request/device/configure_reporting' failed with error: 'Bind 0x286d97000114cdf8/1 manuSpecificSamsungAccelerometer from '0x8c65a3fffe526014/1' failed (Delivery failed for '645'.)'
Info 2025-06-29 13:00:01z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure_reporting', payload '{"data":{},"error":"Bind 0x286d97000114cdf8/1 manuSpecificSamsungAccelerometer from '0x8c65a3fffe526014/1' failed (Delivery failed for '645'.)","status":"error","transaction":"yta18-3"}'
Info 2025-06-29 13:00:02`

I can't figure out how to change their default reporting interval in the Reporting tab of z2m, please can anyone steer me right?

@Morphy99 @aharrys @dustin @Koenkk
